.github/workflows/rust.yml

```diff
@@ -25,6 +25,10 @@ jobs:
       run: sudo apt-get install clang libopenblas-dev libgfortran-10-dev gfortran
     - name: Run tests
       run: OPENBLAS_TARGET=CORE2 cargo test --all
+    - name: Run integration tests
+      run: |
+         ./tests/integration-tests.sh
+      shell: bash
```

Comment:
What do you think about moving it to a separate job? This will allow them to run in parallel.

Reply:
Not sure about it.
I like the idea of a sequential run to fail early the whole chain.

Having to endure the pain of waiting could be a good motivation for us to improve the build time :fire: 

Do you have a strong opinion?

Reply:
I suggested this exactly for having faster feedback. If it is moved into a separate job, in GitHub Actions we will have three success/failure indicators instead of 2. Therefore we will notice early if integration tests failed, instead of seeing this fact only after ordinary tests are ran.

Reply:
And yes, the pipeline is not the fastest right now :smile: . We can maybe add mold linker and caching, both usually improve build times a lot. But this will complicate the pipeline a bit, so I am not exactly sure if we should do it at this stage.

Reply:
I thought about it again and I decided to be more reasonable and to go with your approach.
Would make things effectively better and I believe the pain is present enough already :+1: 
https://github.com/qdrant/qdrant/pull/279/commits/aaf3306d333978f4aee3e84efc3c6650ed01b528